### ISSUE TITLE
`FourierRZToroidalSurface` tangent vectors are always 0 or NaN

### ISSUE THREAD

**Issue by ddudt** [PlasmaControl/DESC#1532]
The compute quantity `n_zeta` is supposed to be in the direction of $e^\zeta = \nabla\zeta$. [In the code](https://github.com/PlasmaControl/DESC/blob/master/desc/compute/_basis_vectors.py#L3436), it is computed as $(e_\rho \times e_\theta) / |e_\rho \times e_\theta|$. But since $e_\rho = 0$ is [hard-coded](https://github.com/PlasmaControl/DESC/blob/master/desc/compute/_surface.py#L156) for the `FourierRZToroidalSurface`, this always evaluates to `n_zeta = 0` or `NaN`. The same issue exists for the quantity `n_theta`, which we compute as $(e_\zeta \times e_\rho) / |e_\zeta \times e_\rho|$. 

For the case of `n_zeta`, I believe the correct implementation should be: 
```
data["n_zeta"] = rpz2xyz_vec(
    jnp.vstack(
        (
            jnp.zeros(transforms["grid"].num_nodes),
            jnp.ones(transforms["grid"].num_nodes),
            jnp.zeros(transforms["grid"].num_nodes),
        )
    ).T,
    phi=data["phi"],
)
```
Assuming $\zeta = \phi$, this is by definition the unit vector in the direction of $\nabla\zeta$, and this implementation would work for all classes. 

The case of `n_theta` is not as simple, but if the directors were orthogonal we could do something like:
```
n_theta = cross(n_rho, n_phi)
data["n_theta"] = n_theta / jnp.linalg.norm(n_theta, axis=-1)[:, None]
```

**Comment by dpanici** [comment]
If you want the surface tangent vector, that should be things like `e_theta` or `e_zeta`, for getting $\frac{d \bm{x}}{d\theta}$ and $\frac{d \bm{x}}{d\zeta}$  on the surface

**Comment by unalmis** [comment]
The cause of this issue is #1127 . 

Generally it is not possible to compute gradient vectors from data sampled over a surface because there is no information the variation of the coordinate in the direction normal to the surface.

The direction of the gradient vectors need not have the same restriction. Unit normals to the surface are known and so are normalized gradients that do not vary with the coordinate that specifies the surface.


$\nabla \phi = R^{-1} \hat{\phi}  = R^{-2} e_{\phi}|_{R, Z}$ is such an example.

The gradients of general curvilinear coordinates are not.

**Comment by dpanici** [comment]
- just use existing grad(phi)
- don't let surfaces that `n_zeta` makes no sense compute `n_zeta` (i.e. remove from `FourierRZToroidalSurface`)
